octopus-server/src/test/java/octopus/teamcity/server/generic/CreateReleaseStepTest.java

```diff
@@ -0,0 +1,105 @@
+package octopus.teamcity.server.generic;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import jetbrains.buildServer.serverSide.InvalidProperty;
+import octopus.teamcity.common.createrelease.CreateReleasePropertyNames;
+import org.junit.jupiter.api.Test;
+
+class CreateReleaseStepTest {
+
+  private final CreateReleaseStep step = new CreateReleaseStep();
+
+  @Test
+  public void describePropertiesProducesExpectedOutput() {
+    String message = step.describeParameters(buildPropertiesMap());
+    assertEquals("Project name: Project-1\nPackage version: 1.0.0", message);
+  }
+
+  @Test
+  public void validatePropertiesReturnsEmptyInvalidPropertiesListWithValidProperties() {
+    List<InvalidProperty> invalidProperties = step.validateProperties(buildPropertiesMap());
```

Comment:
its a bit annoying, but if possible, its worth trying to use 'final' everywhere you can - params and variables
(surprised errorprone didn't pick this up :thinking:

We may not have configured it to do so ...

Reply:
Done